pkg/genlib/fields/load.go

```diff
@@ -61,42 +62,99 @@ func makePackageURL(baseURL, integration, version string) (*url.URL, error) {
 	return u, nil
 }
 
-func getFieldsFiles(ctx context.Context, packageURL *url.URL, dataStream string) ([]byte, error) {
-	body, err := getFromURL(ctx, packageURL.String())
+func makeDownloadURL(baseURL, donwloadPath string) (*url.URL, error) {
+
+	u, err := url.Parse(baseURL)
 	if err != nil {
 		return nil, err
 	}
 
-	var assetsPayload struct {
-		Assets []string `json:"assets"`
+	u.Path = path.Join(u.Path, donwloadPath)
+	return u, nil
+}
+
+func getFieldsFiles(ctx context.Context, baseURL, integration, dataStream, version string) ([]byte, error) {
+	packageURL, err := makePackageURL(baseURL, integration, version)
+	if err != nil {
+		return nil, err
 	}
 
-	if err = json.Unmarshal(body, &assetsPayload); err != nil {
+	r, err := getFromURL(ctx, packageURL.String())
+	if err != nil {
 		return nil, err
 	}
 
-	fieldsFilesURL := make([]string, 0)
-	prefixFieldsPath := path.Join(packageURL.Path, dataStreamSlug, dataStream, fieldsSlug)
-	for _, assetPath := range assetsPayload.Assets {
-		if !strings.HasPrefix(assetPath, prefixFieldsPath) {
-			continue
-		}
+	var downloadPayload struct {
+		Download string `json:"download"`
+	}
 
-		fieldsFilesURL = append(fieldsFilesURL, assetPath)
+	body, err := ioutil.ReadAll(r)
+	if err = json.Unmarshal(body, &downloadPayload); err != nil {
+		return nil, err
 	}
 
+	downloadURL, err := makeDownloadURL(baseURL, downloadPayload.Download)
+	r, err = getFromURL(ctx, downloadURL.String())
+	if err != nil {
+		return nil, err
+	}
+
+	h := sha256.New()
+	h.Write([]byte(downloadURL.String()))
+	prefix := hex.EncodeToString(h.Sum(nil))
+
+	packageTempDir, err := os.MkdirTemp(settings.CacheDir(), prefix)
```

Comment:
Do we need to spool to disk?  Cannot keep zip in RAM?  How big is it?

If continue to spool to disk, perhaps clean up the temp dir on function exit?

Reply:
> Do we need to spool to disk? Cannot keep zip in RAM? How big is it?

I have to check if zip provide an API for opening a reader

Reply:
did it it ram